### PR TITLE
Merge mpi optimization

### PR DESCRIPTION
**Pull request by iaraota**
Merged your optimization with MPI. 

To build:
```
mkdir build
cd build
cmake ..
cmake --build .
make parallel
```

To test and benchmark MPI:
```
cd examples
mpirun -n 4 ../ljmd.x < argon_2916.inp
```
The time is slower for the small 108 system. We can test for different number of processes.

### CODE REVIEW

**Review by neerukulhari — APPROVED** [review]

**Comment by neerukulhari** [comment]
But we can test for mpirun -np 2 for 108, if time is less

File: src/main.c [hunk 1]
```diff
@@ -16,47 +16,59 @@ int main(int argc, char **argv)
...
+    fill_mpi_struct(&sys);
+    
+    if (sys.mpirank==0) printf("LJMD version %3.1f\n", LJMD_VERSION);
 
     t_start = wallclock();
 
     /* read input file */
-    add_data(stdin, &line, &restfile, &trajfile, &ergfile, &sys, &nprint);
-    
+    if (sys.mpirank==0) add_data(stdin, &line, &restfile, &trajfile, &ergfile, &sys, &nprint);
```
Comment (neerukulhari) [inline src/main.c:29]: I think we can print here communication time, just to have idea

Comment (neerukulhari) [reply src/main.c:29]: Why you did add_data again here

File: src/main.c [hunk 2]
```diff
@@ -16,47 +16,59 @@ int main(int argc, char **argv)
...
+            perror("cannot read restart file");
+            return 3;
         }
-        fclose(fp);
-    } else {
-        perror("cannot read restart file");
-        return 3;
     }
 
     /* initialize forces and energies.*/
```
Comment (neerukulhari) [inline src/main.c:56]: should we use MPI_Barrier here just so call returns at any process only after all members of the communicator have entered the call.

**Review by neerukulhari — APPROVED** [review]

File: src/main.c (hunk 1 above)
Comment (iaraota) [reply src/main.c:29]: I just putted the if in front

File: src/main.c (hunk 2 above)
Comment (iaraota) [reply src/main.c:56]: There is a broadcast in the force call, so they will sync there. I dont think the barrier is needed

**Pull request merged**